File: 2335. Minimum Amount of Time to Fill Cups/solution.java
class Solution {
  public int fillCups(int[] amount) {
    int total = 0;
    int highestValue = 0;
    for (int index = 0; index < amount.length; index++){
      int value = amount[index];
      total += value;
      if (value > highestValue){
        highestValue = value;
      }
    }

    int result = highestValue;
    if ((total + 1) / 2 > result){
      result = (total + 1) / 2;
    }

    return result;
  }
}

File: 2335. Minimum Amount of Time to Fill Cups/solution.py
class Solution(object):
  def fillCups(self, amount):
    """
    :type amount: List[int]
    :rtype: int
    """
    total = 0
    highestValue = 0
    for value in amount:
      total += value
      if value > highestValue:
        highestValue = value

    result = highestValue
    if (total + 1) // 2 > result:
      result = (total + 1) // 2

    return result